### PR TITLE
Add checks for erasure, remove assert

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Erasure does not validate blobs it recovers. A similar version needs to be done for shreds

@pgarg66, note that this will also address issue https://github.com/solana-labs/solana/issues/4584

#### Summary of Changes

Fixes #
#5570 
#4584

### CODE REVIEW

File: core/src/blocktree.rs [hunk 1]
```diff
@@ -1741,6 +1734,43 @@ fn handle_recovery(
     }
 }
 
+fn is_recovered_blob_valid(
```
Comment (carllin) [inline core/src/blocktree.rs]: @pgarg66

**Comment by aeyakovenko** [comment]
@carllin @pgarg66 there erasure blobs are signed, right?  so if an attacker tried to corrupt someone else block, they can't just send faulty repairs?

**Comment by pgarg66** [comment]
> @carllin @pgarg66 there erasure blobs are signed, right? so if an attacker tried to corrupt someone else block, they can't just send faulty repairs?

Yes, the erasure blobs are signed, and we check the signature. An attacker should not be able to corrupt the block..

**Comment by carllin** [comment]
@aeyakovenko, yes the erasure blobs are signed. However, any erasure blobs that are recovered through the erasure will not contain the signature, so those cannot be sent via repair to other validators. B/c of this, we don't store erasure blobs recovered through erasure, and hope that Turbine has enough redundancy to propagate these erasure blobs.

**Pull request closed without merging**